### DISCUSSION TITLE
When using a COLUMNS expression, how would you preserve the column names?

### DISCUSSION THREAD

**Discussion by nsiicm0** [duckdb/duckdb#7107] [Ideas]
I find myself frequently writing COLUMNS expressions using regex to select a multitude of columns (100+). https://duckdb.org/docs/sql/expressions/star#columns
This works great. However, once you need to to further processing AFTER the selection, e.g., in a following CTE, it becomes cumbersome, as you cannot specify the column names anymore.

Example:
```python
import pandas as pd
import duckdb

xdf = pd.DataFrame({'a': [1,2,3,None], 'b': [1,2,3,4]}) 
duckdb.query("""
SELECT
    AVG(COALESCE(COLUMNS('(a|b)'), 0)::int)
FROM xdf
""")
```

This will result in a table that looks like this:
```
┌──────────────────────────────────────────┬──────────────────────────────────────────┐
│ avg(CAST(COALESCE(xdf.a, 0) AS INTEGER)) │ avg(CAST(COALESCE(xdf.b, 0) AS INTEGER)) │
│                  double                  │                  double                  │
├──────────────────────────────────────────┼──────────────────────────────────────────┤
│                                      1.5 │                                      2.5 │
└──────────────────────────────────────────┴──────────────────────────────────────────┘
```

Does anyone know of a way to preserve the original column name, or re-set the alias to the original column name?

**Comment by Phantom168** [comment]
Eagerly waiting for the solution to this

**Comment by ericemc3** [comment]
What about a lambda function as a second parameter in COLUMNS()?

```
SELECT
    AVG(COALESCE(COLUMNS('(a|b)', c -> c), 0)::int)
```

**Comment by Mytherin** [comment]
[I've opened a PR](https://github.com/duckdb/duckdb/pull/10774) with improvements to the naming of `COLUMNS(*)`:

```sql
D select columns(*)::VARCHAR FROM (select 42 a, 84 b);
┌─────────┬─────────┐
│    a    │    b    │
│ varchar │ varchar │
├─────────┼─────────┤
│ 42      │ 84      │
└─────────┴─────────┘
D select MIN(COLUMNS(*)) AS "min_\0" FROM (select 42 a, 84 b);
┌───────┬───────┐
│ min_a │ min_b │
│ int32 │ int32 │
├───────┼───────┤
│    42 │    84 │
└───────┴───────┘
```

**Reply by nsiicm0** [reply]
This is brilliant. Thank you!

**Comment by soerenwolfers** [comment]
This is now available via regex syntax, e.g., `COLUMNS('test_(\w+)') AS '\1'` (matching groups are one-indexed, and `\0` is the original column name) so this can be closed.
